### PR TITLE
tests: refactor and rely on matrix-stack only

### PR DESCRIPTION
**Pull request by gaelgatelement**
- Env files now contain the path to the pytest value file
 - All charts deployment now rely on `matrix-stack` to get ready for charts flattening

### CODE REVIEW

File: charts/synapse/values.yaml
Comment (benbz) [inline charts/synapse/values.yaml:1]: Shouldn't be disappearing?

File: tests/fixtures/synapse.py [hunk 1]
```diff
@@ -13,18 +13,19 @@
...
-logging.getLogger("pyhelm3.commands").setLevel(logging.ERROR)
+logging.getLogger("pyhelm3").setLevel(logging.ERROR)
 
 
 @pytest.fixture(scope="session")
 async def synapse_ready(
     cluster, kube_client: AsyncClient, helm_client: pyhelm3.Client, ingress, ess_namespace, generated_data: ESSData
 ):
     counter = 0
+
```
Comment (benbz) [inline tests/fixtures/synapse.py]: Can all the below be updated to use the `revision_deployed` stuff?

File: charts/synapse/values.yaml
Comment (gaelgatelement) [reply charts/synapse/values.yaml:1]: It's going to happen in next PR

Comment (benbz) [reply charts/synapse/values.yaml:1]: I just expect that this disappearing is causing lots of your CI failures

Comment (gaelgatelement) [reply charts/synapse/values.yaml:1]: ah I read "should it be disappearing ?" :facepalm:

File: tests/fixtures/synapse.py (hunk 1 above)
Comment (benbz) [reply tests/fixtures/synapse.py]: Have I misunderstood what `revision_deployed` should do? I'd have thought lines 22-38 could be replaced by it somehow similar to what EW does

Comment (gaelgatelement) [reply tests/fixtures/synapse.py]: oh you are right, this should be simplifyed greatly indeed.

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -92,6 +95,9 @@ jobs:
         poetry install
         echo "$(poetry env info -p)/bin" >> "${GITHUB_PATH}"
 
+    - name: Update helm dependencies recursively
```
Comment (benbz) [inline .github/workflows/build-test.yml]: Why is this needed as well as l68-70?
